Performance Stress Test Implementation

Raxnam opened the pull request:
Implemented stress tests
- CPU Stress Test 
     Automatic core detection or user scalable
     Target specific utilization targets (+- 5%)
- GPU Stress Test
- Memory Utilization Test
     User scalable memory allocation
     Note: The memory once allocated is still held onto by Unity and not returned to the system regardless of it being released by the garbage collector

Commit:
Feat:(Performance Stress Test) CPU Test and Partial GPU Test

added CPU Test and partial working GPU Test.

Commit:
feat(Stress Test) Finished Graphics Stress Test

Finished the graphics stress test and cleaned up the UI

Commit:
Feat(performance test) Finished Graphics Test and memory test

Finished graphics test
- changed some project settings to use the heavier render pipeline and settings
started and finished memory test
Added stress test scene to the login ui prefab

Commit:
Feat(performance stress test) Target utilization slider

Added a target utilization slider for the cpu test

Commit:
docs(performance stress test) Added Readme

Added a readme for the stress test scene

pew-dahs commented on Assets/Scripts/Util/RotateObject.cs:
```diff
@@ -0,0 +1,19 @@
+public class RotateObject : MonoBehaviour
+{
+    public float rotationSpeed = 25.0f;
+    // Start is called before the first frame update
```
If there's nothing to do for Start, should just remove it.

pew-dahs commented on Assets/Scripts/PerformanceStressTest.cs:
```diff
@@ -0,0 +1,206 @@
+    public GaseousArray()
+    {
+        primaryContainer = new List<List<Databomb>>();
+
```
nit: Extra white space

pew-dahs commented on Assets/Scripts/PerformanceStressTest.cs:
```diff
@@ -0,0 +1,206 @@
+            {
+                try
+                {
+                    
```
nit: Extra whitespace

pew-dahs commented on Assets/Scripts/PerformanceStressTest.cs:
```diff
@@ -0,0 +1,206 @@
+        UnityEngine.Debug.Log("Stopping: " + threadList.Count + " threads");
+        for (int i = 0; i < threadList.Count; i++)
+        {
+
```
nit: Extra whitespace

pew-dahs commented on Assets/Materials/New Lighting Settings.lighting:
```diff
@@ -0,0 +1,64 @@
+%YAML 1.1
```
I feel like this file and the shader graph should have proper names instead of "New File."

pew-dahs commented on .gitignore:
```diff
@@ -142,3 +142,326 @@ test-results/
 *IL2CPPCache/*
 *IL2CPPStats/*
 EOSPluginEditorConfiguration/
+Assets/TextMesh Pro/Sprites/EmojiOne.png.meta
```
All the TextMeshPro files can be ignored with `[Aa]ssets/TextMesh*Pro/` instead of adding each file individually.

pew-dahs commented on Assets/Scripts/UI/UISliderToText.cs:
```diff
@@ -0,0 +1,16 @@
+    public Slider sliderObject;
+    public TextMeshProUGUI SliderText;
+
+    public void updateText()
```
nit: You can use dynamic values from a slider event from the Unity property window assuming you're using that, which would eliminate the need to keep track of the slider. However, if you're calling this not from the Unity event, then ignore this.

pew-dahs reviewed (commented):
The biggest thing I see is the large change to the gitignore that could be done with a single line. Everything else is okay other that some nitpicks and file naming.

andrew-hirata-playeveryware commented on Assets/Materials/New Shader Graph.shadergraph:
```diff
@@ -0,0 +1,1803 @@
+{
```
This file could use a proper name.

Commit:
chore(performance stress test) cleaned up code

Cleaned up code and changed how the core slider override worked so it would display the number of cores that unity detects when active

cleaned out the gitignore to just ignore the whole text mesh pro folder instead of each file individually

added the copyright notice to the top of each new file

andrew-hirata-playeveryware reviewed (commented):
We need to add a license file for the images: 

```
PLAYEVERYWARE and [image001.png] are trademarks of PlayEveryWare, Inc. 
Subject to the term and conditions of [these Terms of Use; this Agreement] [Developers] may 
place such Marks on [projects] for the sole purpose of identifying PlayEveryWare as the 
owner and copyright holder of the [PlugIn] and for no other business or non-business 
purposes whatsoever and no other goods or services whatsoever. 
```

dammittom reviewed (changes requested):
Could we rename these files to "PlayEveryWare-logo"? This would better align with company branding guidelines, which would also help enforce other users to comply with them as well.

Commit:
chore(PerformanceStressTEst) Changed file names and added license

Changed image file names to be both more descriptive and fit with other company naming conventions

Added a license file for the company logo usage

andrew-hirata-playeveryware commented on Assets/Materials/PlayEveryWareLicense.txt:
```diff
@@ -0,0 +1,5 @@
+PLAYEVERYWARE, PlayEveryWare-normalmap.png, and PlayEveryWare-logo.png are trademarks of PlayEveryWare, Inc. 
```
Hey @dammittom, Does this look good to you?

dammittom replied:
This looks great! Thank you!

andrew-hirata-playeveryware commented:
Merged in manually to resolve merge conflicts at 2b8c40da91cfb6e17ae448d865a9dc8312f5e031
